maths/binary_exponentiation.py

```diff
@@ -1,20 +1,78 @@
-"""Binary Exponentiation."""
+def binary_exponentiation_multiplication(a, b):
```

Comment:
It looks like this function calculates `a * b` rather than `a ** b`, so shouldn't it belong in a file like `binary_multiplication.py` instead?